plugin/trino-loki/src/main/java/io/trino/plugin/loki/LokiTableHandle.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.loki;
+
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ConnectorTableHandle;
+
+import java.time.Instant;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public record LokiTableHandle(String query, Instant start, Instant end, List<ColumnHandle> columnHandles)
+        implements ConnectorTableHandle
+{
+    public LokiTableHandle
+    {
+        requireNonNull(query);
+        requireNonNull(start);
+        requireNonNull(end);
+        requireNonNull(columnHandles);
```

Comment:
nit: columnHandles = Immutable.copyOf(requireNonNull(columnHandles)

Reply:
How can this be done? It says `columnHandles` is final.